forward/utils.go

```diff
@@ -17,9 +17,27 @@ func cloneURL(i *url.URL) *url.URL {
 // copyHeaders copies http headers from source to destination.  It does not
 // overide, but adds multiple headers
 func copyHeaders(dst, src http.Header) {
+	var extraHopByHopHeaders []string
 	for k, vv := range src {
-		for _, v := range vv {
-			dst.Add(k, v)
+		switch k {
```

Comment:
[`exprSwitch.walkCases`](https://golang.org/src/cmd/compile/internal/gc/swt.go#L286) will generate binary search if there are 4 or more constant cases, but it's nothing important, choosing language feature or data structure is more or less a personal perference.

Another reason I use `switch` here is to avoid lookup the "Connection" header separately.

Does concatenating all cases together, like below, seem more natural to you?

```
case "Keep-Alive", "Proxy-Authenticate", "Proxy-Authorization", "TE", "Trailers", "Transfer-Encoding", "Upgrade":
        continue
```